### PR TITLE
remove non-locale innerHTML calls and add CI check

### PR DESCRIPTION
**Pull request by diracdeltas**
Resolves https://github.com/brave/brave-browser/issues/11686

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

The only new feature that this adds is a CI check for instances of `innerHTML` (not counting the ones set in localization). To check that, run `npm run check_security`. 

This PR has a high risk of UI regressions, so QA should check for this. See https://github.com/brave/brave-browser/issues/11686 for QA test steps.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_extension/extension/brave_extension/webstore.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ const callback = (mutationsList: MutationRecord[], observer: MutationObserver) =
   const buttons: NodeListOf<Element> = document.querySelectorAll('div.webstore-test-button-label')
 
   buttons.forEach((button: Element) => {
-    const text: string = button.innerHTML
+    const text: string = button.textContent || ''
     if (textToMatch.includes(text)) {
-      button.innerHTML = text.replace('Chrome', 'Brave')
+      button.textContent = text.replace('Chrome', 'Brave')
```
Comment (diracdeltas) [inline components/brave_extension/extension/brave_extension/webstore.ts:10]: @keur how do i test that this is working? not sure where the webstore.ts file gets loaded.

Comment (keur) [reply components/brave_extension/extension/brave_extension/webstore.ts:10]: Open any [chrome extension](https://chrome.google.com/webstore/detail/bookmark-manager/idakfiahffeejfhghndaboolmmhbnepn?hl=en-US) and it should say "Add to Brave" and not "Add to Chrome"

You can also see how these are loaded [here](https://github.com/brave/brave-core/blob/deab19b6b5ec76f20de07ac94cd418df78d22f43/components/brave_extension/extension/brave_extension/manifest.json#L47-L52)

Comment (keur) [reply components/brave_extension/extension/brave_extension/webstore.ts:10]: Usually there are two buttons, so make sure it also gets the one at the bottom of the page too.

**Comment by NejcZdovc** [comment]
@diracdeltas please create branch in this repo so that CI can run. Sadly CI doesn't run on forks

**Comment by diracdeltas** [comment]
@NejcZdovc please see https://github.com/brave/brave-core/pull/6646, thanks

**Pull request closed without merging**